pol-core/pol/uworld.cpp

```diff
@@ -99,10 +98,10 @@ void move_multi_in_world( unsigned short oldx, unsigned short oldy, unsigned sho
     newzone.multis.push_back( multi );
   }
 
-  if ( multi->realm() != oldrealm )
+  if ( multi->realm() != newpos.realm() )
```

Comment:
I wonder if there's any side-effect of changing this from oldpos to newpos. Technically before the multi->realm() was already changed before this function was called, now it needs to be called afterwards